src/main/java/com/azure/cosmos/kafka/connect/sink/CosmosDBSinkConnector.java

```diff
@@ -48,9 +62,61 @@ public void stop() {
     public ConfigDef config() {
         return CosmosDBSinkConfig.getConfig();
     }
-    
+
     @Override
     public String version() {
         return this.getClass().getPackage().getImplementationVersion();
     }
+
+    @Override
+    public Config validate(Map<String, String> connectorConfigs) {
+        Config config = super.validate(connectorConfigs);
+        if (config.configValues().stream().anyMatch(cv -> !cv.errorMessages().isEmpty())) {
+            return config;
+        }
+
+        Map<String, ConfigValue> configValues = config.configValues().stream().collect(
+            Collectors.toMap(ConfigValue::name, Function.identity()));
+
+        validateConnection(connectorConfigs, configValues);
+        validateTopicMap(connectorConfigs, configValues);
+
+        return config;
+    }
+
+    private void validateConnection(Map<String, String> connectorConfigs, Map<String, ConfigValue> configValues) {
+        String endpoint = connectorConfigs.get(CosmosDBSinkConfig.COSMOS_CONN_ENDPOINT_CONF);
+        String key = connectorConfigs.get(CosmosDBSinkConfig.COSMOS_CONN_KEY_CONF);
+        try {
+            readAllDatabases(endpoint, key);
```

Comment:
rather than iterating over all databases in the account, we should be able to just create the Cosmos client, the buildClient() method should validate the endpoint and key.